JSDemos/Demos/Tabs/Overview/React/App.tsx

```diff
@@ -43,70 +60,99 @@ const App = () => {
     setIconPosition(e.value);
   }, [setIconPosition]);
 
-  const orientationChanged = React.useCallback(
-    (e) => {
-      setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
-      setOrientation(e.value);
-    }, [setOrientation, setWidgetWrapperClasses],
-  );
+  const orientationChanged = React.useCallback((e) => {
+    setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
+    setOrientation(e.value);
+  }, [setOrientation, setWidgetWrapperClasses]);
 
   const showNavigationChanged = React.useCallback((e) => {
+    const shouldRestrictWidth = e.value || scrollContent;
+
+    toggleShouldRestrictWidth(shouldRestrictWidth);
     setShowNavigation(e.value);
-  }, [setShowNavigation]);
+  }, [scrollContent, setShowNavigation, toggleShouldRestrictWidth]);
 
   const scrollContentChanged = React.useCallback((e) => {
+    const shouldRestrictWidth = e.value || showNavigation;
+
+    toggleShouldRestrictWidth(shouldRestrictWidth);
     setScrollContent(e.value);
-  }, [setScrollContent]);
+  }, [showNavigation, setScrollContent, toggleShouldRestrictWidth]);
 
   const fullWidthChanged = React.useCallback((e) => {
     setFullWidth(e.value);
-  }, [setFullWidth]);
+    setWidth(e.value ? '100%' : 'auto');
+  }, [setFullWidth, setWidth]);
 
   const rtlEnabledChanged = React.useCallback((e) => {
     setRtlEnabled(e.value);
   }, [setRtlEnabled]);
 
+  const selectionChanged1 = React.useCallback((args) => {
+    if (args.name === 'selectedIndex') {
+      setSelectedIndex1(args.value);
+    }
+  },
+  [setSelectedIndex1]);
+
+  const selectionChanged2 = React.useCallback((args) => {
+    if (args.name === 'selectedIndex') {
+      setSelectedIndex2(args.value);
+    }
+  },
+  [setSelectedIndex2]);
+
+  const selectionChanged3 = React.useCallback((args) => {
+    if (args.name === 'selectedIndex') {
+      setSelectedIndex3(args.value);
+    }
+  },
+  [setSelectedIndex3]);
+
   return (
     <div id="tabs-demo">
       <div className="widget-container">
         <div className={widgetWrapperClasses}>
           <Tabs
             id="withText"
-            width="auto"
-            selectedIndex={0}
+            width={width}
+            selectedIndex={selectedIndex1}
             rtlEnabled={rtlEnabled}
             dataSource={tabsText}
             scrollByContent={scrollContent}
             showNavButtons={showNavigation}
             orientation={orientation}
             stylingMode={stylingMode}
             iconPosition={iconPosition}
+            onOptionChanged={selectionChanged1}
```

Comment:
Do we need to add this if we're not using the selectedIndex1 state? To maintain the functionality of changing the selection, using defaultSelectedIndex might suffice

Reply:
These parts was removed in [5b9bd3875eddb26acdca83ce94cfe7a767786467].